### PR TITLE
ARTEMIS-1757 Improve DB2 compatibility

### PR DESCRIPTION
**Pull request by franz1981**
It includes:
- Fixed AUTO_INCREMENT not supported on DB2
- Added proper stop/teardown of driver
- Fixed recursive SQLException formatting

### CODE REVIEW

**Comment by franz1981** [comment]
DO NOT MERGE YET, I'm running some tests to see if there are other evident compatibility issues (SQL syntax and weird exception/driver behaviours).

**Comment by franz1981** [comment]
The tests has worked and it could be merged safely. If new compatibility issues will arise I will push a new PR with the additional fixes :)

File: artemis-jdbc-store/src/main/java/org/apache/activemq/artemis/jdbc/store/drivers/AbstractJDBCDriver.java [hunk 1]
```diff
@@ -95,6 +95,7 @@ public void stop() throws SQLException {
       synchronized (connection) {
          if (sqlProvider.closeConnectionOnShutdown()) {
             try {
+               connection.setAutoCommit(true);
```
Comment (clebertsuconic) [inline artemis-jdbc-store/src/main/java/org/apache/activemq/artemis/jdbc/store/drivers/AbstractJDBCDriver.java]: this is dangerous.. isn't?

why did you need it?
